build/templates/session.py/fancy_self_test.py.mako

```diff
@@ -2,14 +2,19 @@
 <%
     '''Call self-test and throw if there is a failure'''
     import build.helper as helper
+    lock_indent = '    ' if f['use_session_lock'] else ''
```

Comment:
This seems pretty repetitive and dirty.
I would think we can use Python decorators for this stuff. Did you look into that?

Reply:
Decorators can't access the function parameters, which is what we would need to be able to call lock_session and unlock_session from the decorator.
At least not with out relying on parameters being consistent with vi always first and assuming that.
Is that acceptable? If so I can try something based on https://stackoverflow.com/questions/5929107/decorators-with-parameters

Reply:
Can you rely on the first parameter always being `self` always being a `Session`?
Since this may take a little while, I'd like to suggest adding the lock/unlock functions and context manager in one PR, then using it in a separate PR.